expression/builtin_time.go

```diff
@@ -2652,10 +2652,11 @@ func (du *baseDateArithmitical) getIntervalFromString(ctx sessionctx.Context, ar
 }
 
 func (du *baseDateArithmitical) getIntervalFromDecimal(ctx sessionctx.Context, args []Expression, row chunk.Row, unit string) (string, bool, error) {
-	interval, isNull, err := args[1].EvalString(ctx, row)
+	decimal, isNull, err := args[1].EvalDecimal(ctx, row)
```

Comment:
Unexpectedly, we do not have a case to cover this line before. 😢

Reply:
@XuHuaiyu I add unit test for getIntervalFromDecimal PTAL